### PR TITLE
Deprecate `process_position` from the Trainer constructor

### LINKED ISSUE #8968: Deprecate `process_position` from the Trainer constructor
## 🚀 Feature

Remove this two arguments from the Trainer constructor

### Motivation

We are auditing the Lightning components and APIs to assess opportunities for improvements:
- https://docs.google.com/document/d/1xHU7-iQSpp9KJTjI3As2EM0mfNHHr37WZYpDpwLkivA/edit#
- https://github.com/PyTorchLightning/pytorch-lightning/issues/7740

The Trainer today has over 50 arguments to its constructor. This number is growing with each feature release, and makes the trainer cluttered. It also hurts the extensibility of the trainer: a number of arguments passed to the Trainer are to customize other utilities. Plumbing arguments through the trainer creates an undesirable coupling: when the underlying components change, the framework is forced to make breaking API changes in at least two places:
1. The underlying component
2. The trainer constructor

Example: https://github.com/PyTorchLightning/pytorch-lightning/pull/8062
- we now have to change the trainer API in order to address `weights_summary` implementation changes (or deprecate `weights_summary` off the constructor in favor of a callback: https://github.com/PyTorchLightning/pytorch-lightning/issues/8478  )

Example: https://github.com/PyTorchLightning/pytorch-lightning/issues/8780
-`log_gpu_memory` accepting `min_max` is hyper-specific to nvidia-smi, and isn't applicable for torch.cuda.memory stats

Upcoming examples:
- Supporting new types of progress bars https://github.com/PyTorchLightning/pytorch-lightning/pull/8929
- But our current trainer arguments for progress bars could be specific to the TQDM implementation

### Pitch

1. Deprecate `process_position` off the Trainer constructor in v1.5

2. In version 1.7, remove `process_position`from the Trainer entirely

To customize this, users can still construct the ProgressBar callback object and pass it to the Trainer.

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #8968 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM !

File: CHANGELOG.md [hunk 1]
```diff
@@ -172,6 +172,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Updated deprecation of  `argparse_utils.py` from removal in 1.4 to 2.0 ([#9162](https://github.com/PyTorchLightning/pytorch-lightning/pull/9162))
 
 
+- Deprecated passing `process_position` to the `Trainer` constructor ([#9222](https://github.com/PyTorchLightning/pytorch-lightning/pull/9222))
```
Comment (awaelchli) [inline CHANGELOG.md:175]: Let's always mention what the new way is. Let's mention here that process_position has to be set in the ProgressBar callback

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 2]
```diff
@@ -58,6 +59,12 @@ def on_trainer_init(
         self._configure_timer_callback(max_time)
 
         # init progress bar
+        if process_position != 0:
+            rank_zero_deprecation(
+                f"Setting `Trainer(process_position={process_position})` is deprecated in v1.5 and will be removed"
+                " in v1.7. Please pass :class:`~pytorch_lightning.callbacks.progress.ProgressBar` with"
```
Comment (awaelchli) [inline pytorch_lightning/trainer/connectors/callback_connector.py:65]:
```suggestion
                " in v1.7. Please pass `pytorch_lightning.callbacks.progress.ProgressBar` with"
```
this formatting will not do anything for the printed message, this is for sphinx rst format only

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 3]
```diff
@@ -58,6 +59,12 @@ def on_trainer_init(
...
+                " ``process_position`` directly to the Trainer's `callbacks` argument instead."
```
Comment (ethanwharris) [inline pytorch_lightning/trainer/connectors/callback_connector.py:66]: Can also drop the double backticks here as it won't be formatted
```suggestion
                " `process_position` directly to the Trainer's `callbacks` argument instead."
```

**Review by ethanwharris — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (kaushikb11) [reply CHANGELOG.md]: Updated!

**Comment by ananthsub** [comment]
thanks @kaushikb11 !
